features/430-passwordless/implementation.md

```diff
@@ -0,0 +1,40 @@
+# Implementation Guide
+
+## Schema
+
+The provider tables (i.e. identities) will be kept, but password hash in
+password identity will be split out as a new type of authenticator (password
+authenticator).
+
+Existings authenticators for MFA would be kept and compatible with the new
+design. Each authenticator would be associated with a user (i.e. 1 user to
+many authenticators).
+
+## Authentication Session
+
+Authentication sessions would have four kind of steps:
+- Select identity: `identity:select`
+    - A list of usable identities type would be available for selection.
+    - User must provide sufficient information to determine the identity
+      (e.g. login ID), to proceed.
+- Perform primary authentication: `identity:authn`
+    - A list of usable primary authenticator types would be available for
+      selection.
+    - Some of the available authenticator may have not been set up for the
+      identity. These authenticators should always fail the authentication.
+- Perform secondary authentication: `mfa:authn`
+    - A list of usable secondary authenticators (set up for the user) would be
+      available for selection.
+    - At this step, we have sufficient confidence about user's identity to list
+      usable authenticators of the user.
+- Setup secondary authentication: `mfa:setup`
+    - User must setup one secondary authenticator to proceed.
+
+For Auth API, client will always recieve authentication session after
```

Comment:
Not sure what this means. I think this statement is only applicable to app that has MFA turned on?

Reply:
Yes, I should clarify it to be `if client received an authentication session, it'll always have steps after...`.